docs/servo_power_test/src/test_runner.ts

```diff
@@ -28,26 +28,18 @@ export class TestRunner {
     const chunk = await this.dut.read();
     return chunk;
   }
-  public async copyScriptToDut(customScript: string) {
-    const script = `#!/bin/bash -e
-function workload () {
+  public async runWorkload(currentConfigNum: number, customScript: string) {
+    const script = `function workload () {
+  echo "current config: ${currentConfigNum}"
   ${customScript}
 }
 sleep 3
 echo "start"
-workload 1> ./test_out.log 2> ./test_err.log
+workload 1>> ./test_out.log 2>> ./test_err.log
 echo "end"
 sleep 3
 echo "stop"\n`;
-    await this.dut.write('cat > ./example.sh << EOF\n');
-    await this.dut.write(btoa(script) + '\n');
-    await this.dut.write('EOF\n');
-  }
-  public async executeScript() {
-    await this.dut.write('base64 -d ./example.sh | bash\n');
-  }
-  public async executeCommand(s: string) {
-    await this.dut.write(s);
+    await this.dut.write(`echo "${btoa(script)}" | base64 -d | bash\n`);
```

Comment:
Yes! That's exactly what I was suggesting. nice!!!
(maybe adding "\n" at the beginning of the line improves the stability, as I described during our 1:1)